src/client.js

```diff
@@ -32,47 +30,26 @@ function createUppyClient(vm, options = {}) {
   }, options));
 }
 
-const MODE_MAPPING = {
-  TUS: {
-    id: 'Tus',
-    uploader: Tus,
-    options: {
-      headers: { Accept: 'application/json' },
-      limit: 1,
-      resume: false,
-    },
-  },
-  XHR: {
-    uploader: XHR,
-    id: 'XHRUpload',
-  },
-};
-
-export const MODES = Object.keys(MODE_MAPPING);
-
-export const DEFAULT_MODE = 'TUS';
-
 export default class Client {
-  constructor(vm, options = {}) {
+  constructor(vm, uploader, options = {}) {
+    this.uploader = uploader;
     this.uppy = createUppyClient(vm, options.uppy);
-    this.installPlugin(options.uploader, options.mode);
+    this.installPlugin(uploader.uploaderClass, uploader.options);
   }
 
-  installPlugin(options = {}, mode = DEFAULT_MODE) {
-    this.uppy.use(
-      MODE_MAPPING[mode].uploader,
-      Object.assign(MODE_MAPPING[mode].options || {}, options),
-    );
+  installPlugin(plugin, options = {}) {
+    this.uppy.use(plugin, options);
   }
 
-  updateEndpoint(endpoint, { mode = DEFAULT_MODE } = {}) {
-    this.uppy.getPlugin(MODE_MAPPING[mode].id).opts.endpoint = endpoint;
+  updateEndpoint(endpoint) {
```

Comment:
The `reset` function allows to reset a plugin. But if the the plugin being reset is not already defined it will install a new plugin and thus the line `this.uppy.plugins.uploader[0]` will always update the endpoint of the plugin first installed. So either you have to pass a plugin name to `updateEndpoint` or you check if the plugin is already installed in the `reset` method. Both solutions will work for me.

Reply:
👍 I solved it with a static ID for the `vue-drop-zone` plugin. We don't need a dynamic ID because we don't provide multiple plugins.